ballerina/cookie_cookieStore.bal

```diff
@@ -382,21 +383,37 @@ const string URL_TYPE_2 = "http://www.";
 const string URL_TYPE_3 = "http://";
 const string URL_TYPE_4 = "https://";
 
+final regexp:RegExp SLASH_REGEX = re `/`;
+
 // Extracts domain name from the request URL.
 isolated function getDomain(string url) returns string {
     string domain = url;
     if url.startsWith(URL_TYPE_1) {
-        domain = url.substring(URL_TYPE_1.length(), url.length());
+        domain = SLASH_REGEX.split(url.substring(URL_TYPE_1.length(), url.length()))[0];
     } else if url.startsWith(URL_TYPE_2) {
-        domain = url.substring(URL_TYPE_2.length(), url.length());
+        domain = SLASH_REGEX.split(url.substring(URL_TYPE_2.length(), url.length()))[0];
     } else if url.startsWith(URL_TYPE_3) {
-        domain = url.substring(URL_TYPE_3.length(), url.length());
+        domain = SLASH_REGEX.split(url.substring(URL_TYPE_3.length(), url.length()))[0];
     } else if url.startsWith(URL_TYPE_4) {
-        domain = url.substring(URL_TYPE_4.length(), url.length());
+        domain = SLASH_REGEX.split(url.substring(URL_TYPE_4.length(), url.length()))[0];
     }
     return domain.toLowerAscii();
 }
 
+// Construct the request path from client url and the path from client method invocation.
+isolated function getReqPath(string url, string path) returns string {
```

Comment:
we could use regex here.
```
ballerina
function getReqPath(string url, string path) returns string {
    regexp:Groups? groups = re `^(?:https?://)?[^/]+(/.*)?$`.findGroups(url);
    if groups is () || groups.length() == 1 || groups[1] is () {
        return path;
    }
    return (<regexp:Span>groups[1]).substring() + path;
}
```

Reply:
Fixed via [0ae35c3](https://github.com/ballerina-platform/module-ballerina-http/pull/2091/commits/0ae35c346e351278c50a5c9e32c7b32393a4c1c5)